### ISSUE TITLE
aws_iam_openid_connect_provider rejects valid "url"s

### ISSUE THREAD

**Issue by SKalt** [hashicorp/terraform-provider-aws#26483]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v1.1.9
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.27.0
```

### Affected Resource(s)

* aws_iam_openid_connect_provider

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
variable "my_bitbucket_org" {
  type        = string
  description = "the name of my bitbucket org"
  default     = "myorg"
}
provider "aws" {
  region = "us-east-1"
}
locals {
  oidc_url = "api.bitbucket.org/2.0/workspaces/${var.my_bitbucket_org}/pipelines-config/identity/oidc"
}
resource "aws_iam_openid_connect_provider" "bitbucket_pipelines_oidc" {
  url             = local.oidc_url
  thumbprint_list = [] # TODO
  client_id_list  = [] # TODO
}
```

### Debug Output

[plan debug logs](https://gist.github.com/SKalt/5ac624271c1cb0988ddcf1c6f6656ec3)

### Expected Behavior

the provider should have accepted the protocol-less "url" that BitBucket provides.  If I were to enter the same protocol-less url in the AWS IAM console, the OIDC-provider resource would be created successfully.

### Actual Behavior

The resource marks my "url" as invalid, despite the fact that the "url" is a valid _iss_ claim.

### Steps to Reproduce

1. `terraform plan`

**Comment by awolski** [comment]
I hit the same _issue_when setting up [GitHub OIDC provider](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services). I don't think this is an issue though.

I think your problem is that you haven't specified `https://` at the start of your URL. The error is misleading as it states that `host` is expected:

```shell
Error: expected "url" to have a host, got api.bitbucket.org/2.0/workspaces/myorg/pipelines-config/identity/oidc
```

I got a similar issue trying to use `token.actions.githubusercontent.com` as the resource url:

```tf
Error: expected "url" to have a host, got token.actions.githubusercontent.com
│  ...
│  445:   url = "token.actions.githubusercontent.com"
```

Changing this to `url = "https://token.actions.githubusercontent.com"` resolved the error.
```

**Comment by Rohlik** [comment]
I have same issue during the `import` procedure:
```thc
import {
  to = aws_iam_openid_connect_provider.default
  id = "arn:aws:iam::123456789012::oidc-provider/oidc.eks.eu-central-1.amazonaws.com/id/XXXXXXXXXXXXXXXXXXX"
}
```
But that resource doesn't contain `https://`, so it failed on:
```
Error: expected "url" to have a host, got oidc.eks.eu-central-1.amazonaws.com/id/XXXXXXXXXXXXXXXXXXX
```

Please fix it 🙏🏼.

**Comment by SteveByerly** [comment]
I don't think this is the fault of the Terraform provider. This happens because the AWS [`CreateOpenIDConnectProvider` api](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateOpenIDConnectProvider.html#:~:text=The%20URL%20must%20begin%20with) requires the url to start with `https://`.

> The URL must begin with https:// and should correspond to the iss claim in the provider's OpenID Connect ID tokens.

Subsequently, the AWS [`GetOpenIDConnectProvider` api](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetOpenIDConnectProvider.html) returns a response where the `Url` property does _not_ include the `https://` scheme.

[Example response from AWS cli docs:](https://docs.aws.amazon.com/cli/latest/reference/iam/get-open-id-connect-provider.html#examples)

```json
{
    "Url": "server.example.com",
    "CreateDate": "2015-06-16T19:41:48Z",
    "ThumbprintList": [
        "12345abcdefghijk67890lmnopqrst987example"
    ],
    "ClientIDList": [
        "example-application-ID"
    ]
}
```

**Comment by LDVSOFT** [comment]
Yep, I've found that there is a diff suppression function that just overrides the schema. That works for all cases where displaying the stored in the state string isn't needed (i.e. I can't think of any case besides import generating resources), but in my barely relevant opinion normalizing a URL on read would be a clearer way.
